unitTests/controllers/FacetSliderQueryControllerTest.ts

```diff
@@ -62,35 +63,71 @@ export function FacetSliderQueryControllerTest() {
-        controller.putGroupByIntoQueryBuilder(builder);
-        requestForFullRange = builder.groupByRequests[controller.groupByRequestForFullRange];
+        builder = new QueryBuilder();
+        (controller.facet.getSliderBoundaryForQuery as jasmine.Spy).and.returnValue([25, 50]);
       });
 
-      it('should use nosort', () => {
-        expect(requestForFullRange.sortCriteria).toBe('nosort');
-      });
+      describe('with an empty query to clone', () => {
+        beforeEach(() => {
+          controller.putGroupByIntoQueryBuilder(builder);
+          requestForFullRange = builder.groupByRequests[controller.groupByRequestForFullRange];
+        });
+
+        it('should use nosort', () => {
+          expect(requestForFullRange.sortCriteria).toBe('nosort');
+        });
+
+        it('should use the same field', () => {
+          expect(requestForFullRange.field).toBe(facet.options.field);
+        });
+
+        it('should request only one value', () => {
+          expect(requestForFullRange.maximumNumberOfValues).toBe(1);
+        });
 
-      it('should use the same field', () => {
-        expect(requestForFullRange.field).toBe(facet.options.field);
+        it('should not use any query override', () => {
+          expect(requestForFullRange.queryOverride).toBeUndefined();
+        });
+
+        it('should not use any constant query override', () => {
+          expect(requestForFullRange.constantQueryOverride).toBeUndefined();
+        });
+
+        it('should not use any advanced query override', () => {
+          expect(requestForFullRange.advancedQueryOverride).toBeUndefined();
+        });
       });
 
-      it('should request only one value', () => {
-        expect(requestForFullRange.maximumNumberOfValues).toBe(1);
+      it('should use the advanced query override passed in as an option on the component', () => {
+        controller.facet.options.queryOverride = 'an override';
+
+        controller.putGroupByIntoQueryBuilder(builder);
+        requestForFullRange = builder.groupByRequests[controller.groupByRequestForFullRange];
+        expect(requestForFullRange.advancedQueryOverride).toContain('an override');
       });
 
-      it('should set @uri in the advanced query override', () => {
-        expect(requestForFullRange.advancedQueryOverride).toBe('@uri');
+      it('should add the constant query override from the existing query', () => {
+        builder.constantExpression.add('a constant expression');
+        controller.putGroupByIntoQueryBuilder(builder);
+        requestForFullRange = builder.groupByRequests[controller.groupByRequestForFullRange];
+        expect(requestForFullRange.constantQueryOverride).toContain('a constant expression');
       });
 
-      it('should not use any constant query override', () => {
-        expect(requestForFullRange.constantQueryOverride).toBeUndefined();
+      it('should not use any query override if there is none set as an option on the component', () => {
```

Comment:
Though it does not hurt to keep, I think this test could be removed because it looks the same as that on line 104. I see the essence of both tests as being: the value of `requestForFullRange.advancedQueryOverride` should match `facet.options.queryOverride`.